logstash-core/src/main/java/org/logstash/config/ir/graph/DiffResult.java

```diff
@@ -0,0 +1,75 @@
+package org.logstash.config.ir.graph;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Created by andrewvc on 12/14/16.
+ */
+public class DiffResult {
+    private final List<Vertex> removedVertices;
+    private final List<Vertex> addedVertices;
+    private final Collection<Edge> removedEdges;
+    private final Collection<Edge> addedEdges;
+
+    public Collection<Edge> getRemovedEdges() {
+        return removedEdges;
+    }
+
+    public Collection<Edge> getAddedEdges() {
+        return addedEdges;
+    }
+
+    public DiffResult(Collection<Edge> removedEdges, Collection<Edge> addedEdges, List<Vertex> removedVertices, List<Vertex> addedVertices) {
+        this.removedEdges = removedEdges;
+        this.addedEdges = addedEdges;
+        this.removedVertices = removedVertices;
+        this.addedVertices = addedVertices;
+    }
+
+    public String summary() {
+        String template = "(-%d,+%d Edges | -%d,+%d Vertices)";
```

Comment:
There's no need in Java due to the [String Pool](http://stackoverflow.com/questions/2486191/what-is-the-java-string-pool-and-how-is-s-different-from-new-strings)